packages/contracts-bedrock/test/L2/CrossL2Inbox.t.sol

```diff
@@ -0,0 +1,358 @@
+    }
+
+    /// @dev Sets log index in transient storage.
+    /// @param _logIndex Log index to set.
+    function setLogIndex(uint256 _logIndex) external {
+        TransientContext.set(LOG_INDEX_SLOT, _logIndex);
+    }
+
+    /// @dev Sets timestamp in transient storage.
+    /// @param _timestamp Timestamp to set.
+    function setTimestamp(uint256 _timestamp) external {
+        TransientContext.set(TIMESTAMP_SLOT, _timestamp);
+    }
+
+    /// @dev Sets chain ID in transient storage.
+    /// @param _chainId Chain ID to set.
+    function setChainId(uint256 _chainId) external {
+        TransientContext.set(CHAINID_SLOT, _chainId);
+    }
+}
+
+/// @title CrossL2InboxTest
+/// @dev Contract for testing the CrossL2Inbox contract.
+contract CrossL2InboxTest is Test {
+    /// @dev Selector for the `isInDependencySet` method of the L1Block contract.
+    bytes4 constant L1BlockIsInDependencySetSelector = bytes4(keccak256("isInDependencySet(uint256)"));
+
+    /// @dev CrossL2Inbox contract instance.
+    CrossL2Inbox crossL2Inbox;
+
+    /// @dev Sets up the test suite.
+    function setUp() public {
+        // Deploy the L2ToL2CrossDomainMessenger contract
+        vm.etch(Predeploys.CROSS_L2_INBOX, address(new CrossL2InboxWithModifiableTransientStorage()).code);
+        crossL2Inbox = CrossL2Inbox(Predeploys.CROSS_L2_INBOX);
+    }
+
+    /// @dev Tests that the `executeMessage` function succeeds.
+    function testFuzz_executeMessage_succeeds(
+        ICrossL2Inbox.Identifier memory _id,
+        address _target,
+        bytes calldata _message,
+        uint256 _value
+    )
+        external
+        payable
+    {
+        // Ensure that the id's timestamp is valid (less than or equal to the current block timestamp)
+        _id.timestamp = bound(_id.timestamp, 0, block.timestamp);
+
+        // Ensure that the target call is payable if value is sent
+        if (_value > 0) assumePayable(_target);
+
+        // Ensure that the target call does not revert
+        vm.mockCall({ callee: _target, msgValue: _value, data: _message, returnData: abi.encode(true) });
+
+        // Ensure that the chain ID is in the dependency set
+        vm.mockCall({
+            callee: Predeploys.L1_BLOCK_ATTRIBUTES,
+            data: abi.encodeWithSelector(L1BlockIsInDependencySetSelector, _id.chainId),
+            returnData: abi.encode(true)
+        });
+
+        // Ensure that the contract has enough balance to send with value
+        vm.deal(address(this), _value);
+
+        // Look for the call to the target contract
+        vm.expectCall({ callee: _target, msgValue: _value, data: _message });
+
+        // Call the executeMessage function
+        crossL2Inbox.executeMessage{ value: _value }({ _id: _id, _target: _target, _message: _message });
+
+        // Check that the Identifier was stored correctly, but first we have to increment the call depth to the one
+        // where the Identifier is stored in transient storage
```

Comment:
This makes sense, perhaps good to comment why we need to increment, its that `executeMessage` increments + decrements so we need to increment to have the getters use the right context